libs/execution_context/ebpf_program.c

```diff
@@ -1026,36 +1049,53 @@ ebpf_program_invoke(_In_ const ebpf_program_t* program, _Inout_ void* context, _
         }
     }
 
-    ebpf_assert_success(ebpf_state_store(_ebpf_program_state_index, 0));
+Done:
+    if (program_state_stored) {
+        ebpf_assert_success(ebpf_state_store(_ebpf_program_state_index, 0));
+    }
+    if (rundown_protection_acquired) {
+        ebpf_rundown_protection_release(&program->rundown.protection);
+    }
 }
 
 static ebpf_result_t
 _ebpf_program_get_helper_function_address(
-    _In_ const ebpf_program_t* program, const uint32_t helper_function_id, uint64_t* address)
+    _Inout_ ebpf_program_t* program, const uint32_t helper_function_id, uint64_t* address)
 {
     ebpf_result_t return_value;
     void* function_address;
     EBPF_LOG_ENTRY();
 
+    bool rundown_protection_acquired = false;
+
+    if (!ebpf_rundown_protection_acquire(&program->rundown.protection)) {
+        EBPF_RETURN_RESULT(EBPF_INVALID_ARGUMENT);
```

Comment:
Maybe we should be adding a tracelog as well? IMO, going forward, _new_ error handling code should include tracelogging as well (where-ever it would be helpful).  For high frequency instances, a single line comment indicating that would suffice.